datatypes/src/raster/geo_transform.rs

```diff
@@ -175,6 +154,61 @@ impl GeoTransform {
         }
     }
 
+    /// compute the index of the upper left pixel that is contained in the `partition`
+    pub fn upper_left_pixel_idx(&self, partition: &SpatialPartition2D) -> GridIdx2D {
+        // choose the epsilon relative to the pixel size
+        const EPSILON: f64 = 0.000_001;
+        let epsilon: Coordinate2D =
+            (self.x_pixel_size() * EPSILON, self.y_pixel_size() * EPSILON).into();
+
+        let upper_left_coordinate = partition.upper_left() + epsilon;
+
+        self.coordinate_to_grid_idx_2d(upper_left_coordinate)
+    }
+
+    /// compute the index of the lower right pixel that is contained in the `partition`
+    pub fn lower_right_pixel_idx(&self, partition: &SpatialPartition2D) -> GridIdx2D {
+        // as the lower right coordinate is not included in the partition we subtract an epsilon
+        // in order to not include the next pixel if the lower right coordinate is exactly on the
+        // edge of the next pixel
+
+        // choose the epsilon relative to the pixel size
+        const EPSILON: f64 = 0.000_001;
```

Comment:
Fails if the pixel size is less than this, right?

Reply:
no it is multiplied with the resolution in the next step. this is not new i just moved it from tiling_strategy